worker/src/main/scala/org/apache/celeborn/service/deploy/worker/FetchHandler.scala

```diff
@@ -348,7 +353,13 @@ class FetchHandler(
   def handleEndStreamFromClient(streamId: Long, streamType: StreamType): Unit = {
     streamType match {
       case StreamType.ChunkStream =>
-        val (shuffleKey, fileName) = chunkStreamManager.getShuffleKeyAndFileName(streamId)
+        // LocalPartitionReader/DfsPartitionReader could not get shuffle key and disk file via stream id
+        // which does not register stream in ChunkStreamManager.
+        // Meanwhile, WorkerPartitionReader could get shuffle key and disk file from ChunkStreamManager
+        // because of stream registration of ChunkStreamManager.
+        val (shuffleKey, fileName) =
+          if (streams.containsKey(streamId)) streams.remove(streamId)
+          else chunkStreamManager.getShuffleKeyAndFileName(streamId)
```

Comment:
Is it better to modify chunkStreamManager.getShuffleKeyAndFileName method? For example, return null when getShuffleKeyAndFileName